FEAT: Customer 회원가입, 수정 Controller / Service 일부 구현.

JuneParkCode opened the pull request:
## 변경 내용
- 코드 전체적으로 Reformat
- Customer 에서 회원가입 / 수정에 대한 Controller / Service 작성

JuneParkCode committed:
chore: reformat codes

JuneParkCode committed:
chore: reformat codes

JuneParkCode committed:
feat: Customer Controller / Service / Repository

- Repository 에서 필요한 method 를 정의함
- Service / Controller 에서 회원가입, 회원 수정을 구현함.

kimminkyeu commented on src/main/java/com/ticketwar/ticketwar/customer/entity/Customer.java:
```diff
@@ -1,27 +1,47 @@
 
-@Entity(name = "CUSTOMER")
+@Entity
+@Table(name = "CUSTOMER")
```
memo : entity name은 JPQL 사용 편의를 위해 따로 설정하지 않는다.

kimminkyeu commented on src/main/java/com/ticketwar/ticketwar/customer/dto/CustomerDto.java:
```diff
@@ -1,15 +0,0 @@
-import lombok.*;
-
-@Getter
-public class CustomerDto {
```
memo : mapper 혹은 toEntity 인터페이스 상속받아서 하는 것으로

POKUDING commented on src/main/java/com/ticketwar/ticketwar/customer/dto/CustomerReqDto.java:
```diff
@@ -0,0 +1,29 @@
+    this.email = email;
+  }
+
+  public Customer toCustomer() {
```
sungj: toEntity() 를 갖고있는 DTO Interface 를 사용하는것도 좋을거같다

kimminkyeu commented on src/test/java/com/ticketwar/ticketwar/customer/service/CustomerServiceTest.java:
```diff
@@ -0,0 +1,234 @@
+
+
+@SpringBootTest
+@Transactional
```
memo: Test에서 @Transcational을 붙히면 commit이 안되고 rollback된다.

kimminkyeu commented on src/main/java/com/ticketwar/ticketwar/customer/service/CustomerService.java:
```diff
@@ -1,28 +1,101 @@
     }
+    customer.setNickname(updateNickname);
+    customer.setEmail(updateEmail);
+    customerRepository.save(customer);
```
memo : transactional 붙었기 때문에 save 필요 없음

kimminkyeu commented on src/main/resources/yaml/local.yml:
```diff
@@ -15,11 +15,11 @@ spring:
+  #        useLegacyDatetimeCode: false
   datasource:
-    url: jdbc:h2:mem:db;DB_CLOSE_DELAY=-1;MODE=MySQL
+    url: jdbc:h2:file:~/h2/h2db;AUTO_SERVER=true
```
memo : h2 커넥션 1개밖에 못열어서 intellij에서 보는 방식으로 하기 위해 설정

kimminkyeu commented on src/main/java/com/ticketwar/ticketwar/customer/service/CustomerService.java:
```diff
@@ -1,28 +1,101 @@
+   * @param email
+   * @return 존재할 경우 true
+   */
+  private boolean isDuplicateEmail(Long id, String email) {
```
repository의 isExist 메소드 사용하는 방식이 더 낫지 않을지

JuneParkCode commented on src/main/java/com/ticketwar/ticketwar/customer/service/CustomerService.java:
```diff
@@ -1,28 +1,101 @@
+    if (isDuplicateEmail(email)) {
+      throw new BadRequestException("Duplicated email");
+    }
+    Customer newCustomer = customerReqDto.toCustomer();
```
final 붙여야함.

kimminkyeu commented on src/main/java/com/ticketwar/ticketwar/customer/service/CustomerService.java:
```diff
@@ -1,28 +1,101 @@
+   * @param email
+   * @return 존재할 경우 true
+   */
+  private boolean isDuplicateEmail(String email) {
```
추가 규칙
nullable parameter = Optional 사용 필수
Optional이 안붙으면, 그것은 null 이 들어오지 않는다는 것을 확정

JuneParkCode committed:
refactor: transaction 으로 필요하지 않은 .save() 제거

JuneParkCode committed:
remove: 사용하지 않는 isDuplicate* repository method 제거

JuneParkCode committed:
refactor: final 붙지 않은 부분 수정

JuneParkCode committed:
refactor: BaseDto interface 생성 및 CustomerReqDto 적용

JuneParkCode committed:
chore: entity에 equals(), hashCode() 적용

kimminkyeu reviewed (approved):
옆에서 코드 보며 리뷰 진행하였습니다. 이 구조 참고해서 구현 진행하겠습니다. 고생하셨습니다!

JuneParkCode committed:
feat: CustomerMapper 추가 및 Customer Read 관련 controller method 추가

POKUDING reviewed (approved):
함께 코드를 보며 리뷰를 진행하였습니다. 고생많으셨습니다.

JuneParkCode committed:
chore: Customer builder에 id 추가.

JuneParkCode committed:
refactor: CustomerService.signUp, update

- 명확한 테스트, 검증을 위해 엔티티 객체를 반환하도록 변경함.

JuneParkCode committed:
refactor: CustomerQueryService 로 책임 분리

- CustomerService 에서의 isDuplicate... method 의 책임을 CustomerQueryService 로 옮김.
- CustomerQueryService 는 필요한 데이터 처리 요청 로직에 대한 역할을 분담함.

POKUDING reviewed (approved):
CustomerService에서 CustomerQueryService로의 책임 분리를 하는 구조가 Test 및 책임관리에 더 효율적인 것에 동의합니다.
